src/tmpfiles/tmpfiles.c

```diff
@@ -208,20 +216,148 @@ static const Specifier specifier_table[] = {
         {}
 };
 
+static int specifier_kernel_isol(char specifier, const void *data, const void *userdata, char **ret) {
+        if (!empty_or_root(arg_root))
+                return -ENXIO;
+
+        return specifier_kernel_release(specifier, data, userdata, ret);
+}
+
+static int specifier_arch_isol(char specifier, const void *data, const void *userdata, char **ret) {
+        if (!empty_or_root(arg_root))
+                return -ENXIO;
+
+        return specifier_architecture(specifier, data, userdata, ret);
+}
+
+static int id128_read_isol(const char *prefix, const char *p, Id128Format f, sd_id128_t *ret) {
+        _cleanup_free_ char *fullpath = NULL;
+
+        fullpath = path_join(arg_root, p);
+        if (!fullpath)
+                return -ENOMEM;
+        return id128_read(fullpath, f, ret);
+}
+
+static int specifier_machine_id_isol(
+        const char *prefix, char specifier, const void *data, const void *userdata, char **ret) {
+        int r;
+        char *n;
+        sd_id128_t machine_id = {};
+
+        r = id128_read_isol(prefix, "/etc/machine-id", ID128_PLAIN, &machine_id);
```

Comment:
this must use chase_symlinks() so that this follows symlinks relative to the specified root.

Also if arg_root is not set this really should be shortcut to use sd_id128_get_machine(), as before, since that is cached and so on...

Reply:
now using chase_symlinks_and_open